src/main/java/com/shapesecurity/functional/F.java

```diff
@@ -47,5 +47,11 @@
     public default <C> F<C, R> compose(@NotNull final F<C, A> f) {
         return c -> this.apply(f.apply(c));
     }
+
+
+    @NotNull
+    public default <B> F<A, B> then(@NotNull final F<R, B> f) {
```

Comment:
Because I can do this:

```java
maybeJson.map(
  asObject()
    .then(getField("x"))
    .then(asArray())
    .then(getIndex(0))
    .then(asString()))
```

Moreover, `a.composeRight(b)` means exactly the same thing as `a.compose(b)` to me. A proper "obvious" name will be `composedBy`. At least, `flippedCompose` or `composeFlipped`.

Maybe we can synonym this `bind` or `flatMap` as well, but I like `then` quite a lot.